pkg/scheduler/plugins/deviceshare/device_cache.go

```diff
@@ -0,0 +1,338 @@
+	podRequest = quotav1.Mask(podRequest, deviceResourceNames[deviceType])
+	nodeDeviceTotal := n.deviceTotal[deviceType]
+	if len(nodeDeviceTotal) <= 0 {
+		return fmt.Errorf("node does not have enough %v", deviceType)
+	}
+
+	var deviceAllocations []*apiext.DeviceAllocation
+
+	if isMultipleCommonDevicePod(podRequest, deviceType) {
+		var commonDeviceWanted int64
+		var podRequestPerCard corev1.ResourceList
+		switch deviceType {
+		case schedulingv1alpha1.RDMA:
+			commonDevice := podRequest[apiext.KoordRDMA]
+			commonDeviceWanted = commonDevice.Value() / 100
+			podRequestPerCard = corev1.ResourceList{
+				apiext.KoordRDMA: *resource.NewQuantity(commonDevice.Value()/commonDeviceWanted, resource.DecimalSI),
+			}
+		case schedulingv1alpha1.FPGA:
+			commonDevice := podRequest[apiext.KoordFPGA]
+			commonDeviceWanted = commonDevice.Value() / 100
+			podRequestPerCard = corev1.ResourceList{
+				apiext.KoordFPGA: *resource.NewQuantity(commonDevice.Value()/commonDeviceWanted, resource.DecimalSI),
+			}
+		}
+		satisfiedDeviceCount := 0
+		for minor, resources := range n.deviceFree[deviceType] {
+			if satisfied, _ := quotav1.LessThanOrEqual(podRequestPerCard, resources); satisfied {
+				satisfiedDeviceCount++
+				deviceAllocations = append(deviceAllocations, &apiext.DeviceAllocation{
+					Minor:     int32(minor),
+					Resources: podRequestPerCard,
+				})
+			}
+			if satisfiedDeviceCount == int(commonDeviceWanted) {
+				allocateResult[deviceType] = deviceAllocations
+				return nil
+			}
+		}
+		klog.V(5).Infof("node resource doest not satisfy pod's multiple %v request, expect %v, got %v", deviceType, commonDeviceWanted, satisfiedDeviceCount)
+		return fmt.Errorf("node does not have enough %v", deviceType)
+	}
+
+	for minor, resources := range n.deviceFree[deviceType] {
+		if satisfied, _ := quotav1.LessThanOrEqual(podRequest, resources); satisfied {
+			deviceAllocations = append(deviceAllocations, &apiext.DeviceAllocation{
+				Minor:     int32(minor),
+				Resources: podRequest,
+			})
+			allocateResult[deviceType] = deviceAllocations
+			return nil
+		}
+	}
+	klog.V(5).Infof("node resource doest not satisfy pod's %v request", deviceType)
+	return fmt.Errorf("node does not have enough %v", deviceType)
+}
+
+func (n *nodeDevice) tryAllocateGPU(podRequest corev1.ResourceList, allocateResult apiext.DeviceAllocations) error {
+	podRequest = quotav1.Mask(podRequest, deviceResourceNames[schedulingv1alpha1.GPU])
+	nodeDeviceTotal := n.deviceTotal[schedulingv1alpha1.GPU]
+	if len(nodeDeviceTotal) <= 0 {
+		return fmt.Errorf("node does not have enough GPU")
+	}
+
+	var activeMinor int
+	for i := range nodeDeviceTotal {
+		activeMinor = i
+		break
+	}
+
+	if gpuMem, ok := podRequest[apiext.GPUMemory]; ok {
+		podRequest[apiext.GPUMemoryRatio] = memBytesToRatio(gpuMem, nodeDeviceTotal[activeMinor][apiext.GPUMemory])
```

Comment:
a  instruction about GPUMemoryRatio and GPUMemory is needed here, or give a doc link.